iidm/iidm-scripting/src/main/groovy/com/powsybl/iidm/network/scripting/ShuntCompensatorExtension.groovy

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.powsybl.iidm.network.scripting
+
+import com.powsybl.commons.PowsyblException
+import com.powsybl.iidm.network.ShuntCompensator
+import com.powsybl.iidm.network.ShuntCompensatorLinearModel
+import com.powsybl.iidm.network.ShuntCompensatorModelType
+import com.powsybl.iidm.network.ShuntCompensatorNonLinearModel
+
+/**
+ *
+ * @author Miora Ralambotiana <miora.ralambotiana at rte-france.com>
+ */
+class ShuntCompensatorExtension {
+
+    static double getMaximumB(ShuntCompensator self) {
+        if (ShuntCompensatorModelType.LINEAR == self.getModelType()) {
+            // forward bPerSection * maximumSectionCount for linear shunts
+            return self.getModel(ShuntCompensatorLinearModel.class).getbPerSection() * self.getModel().getMaximumSectionCount()
```

Comment:
Is it a bad idea to keep the getMaximumB in all shunt model and provide a convenient function?

Reply:
The issue is that we used to have `getMaximumB` for linear shunts which returned the susceptance of the highest section (not the actual maximum susceptance) so if we put it in the shunt model, we would have several possibilities:
- either keep this behavior for linear shunt but for non linear shunt it returns the actual maximum susceptance
- or change the behavior and in any case, it returns the actual maximum susceptance
- or keep this behavior for linear shunt and for non linear shunt it returns the susceptance of the highest section too

In any case, I find it too confusing... What do you think?

Reply:
My opinion would be that it should return the actual maximum susceptance regardless of the model, as indicated by its name. Maybe another function could return the susceptance of the highest section for linear shunts if needed.

Reply:
I agree with you, but isn't it deceitful to change the behavior of the method? Or I guess we could say that the previous one was buggy as when it was first implemented, it didn't take into account the case where the susceptance per section was negative...